apps/sms/views/shared/js/startup.js

```diff
@@ -105,8 +104,13 @@ var Startup = exports.Startup = {
       // dispatch contentInteractive when all the modules initialized
       SilentSms.init();
 
-      ActivityHandler.init();
-      SystemMessageHandler.init();
+      // When app is run as inline activity we don't want to handle any other
+      // system messages other than "activity" system message.
+      if (navigator.mozHasPendingMessage('activity')) {
```

Comment:
IMO if we can keep code dealing with mozhaspendingmessage (that should go away eventually) in single place it will be easier to maintain all the tricky cases that are happening on startup (though I haven't read bug 1192263 to understand if it's about startup) :)